### ISSUE TITLE
Jump Height Effects Not Working

### ISSUE THREAD

**Issue by Invadermonky** [asanetargoss/HardcoreAlchemy#11]
Jump height increasing effects are broken in-game. Testing shows that these include Boots of the Traveler and Living Armor Strong Legs upgrade.

This bug is caused by `B:"Controllable Jump Boost"=true` in `potioncore.cfg`. Setting this to false fixes the issue.

On another note, I coded a JER integrations mod for the pack. I haven't posted it anywhere but if you would like I can give you the source if you want to include it in the pack.

**Comment by Invadermonky** [comment]
Current Source and v1.0.0 build.
Integrations:
- Ad Inferos (Mobs, Crops)
- Ars Magica (Dungeons, Mobs)
- Astral Sorcery (Shrines)
- Embers (Mobs)
- Ender Zoo (Mobs)
- Havestcraft
- Mystical Agriculture (Crops)
- Thaumcraft (Mobs)
- Unique Crops (Crops)

Code for Harvestcraft integration is included but the crop models are not working correctly in JER for some reason so the code is commented out.

[HcAJER.zip](https://github.com/asanetargoss/HardcoreAlchemy/files/15450084/HcAJER.zip)

**Comment by asanetargoss** [comment]
Thanks for the bug report. Not sure what "Controllable Jump Boost" does, but if it's for some potion that Potion Core adds, then ideally there should be some compat that allows the different mods to control the jump boost without interfering with each other. But I'm glad the config option workaround is working for you.

I am looking at `HcAJER.zip`, and it looks like it could be a good addition to the pack, but here are my thoughts:

- What is your preferred license for the mod?
- I notice this mod adds compat for a variety of mods. The way it is written, it could crash if some of the referenced mods are not loaded, but I haven't tested this yet. My currently preferred method for getting around a `NoClassDefFoundError` crash involving a third-party mod: is to conditionally call a static function in an inner class named after the mod it's interfacing with. To be clear, I haven't tested this yet, so I'm not sure if it's an issue. I wouldn't reject the mod's inclusion on that basis alone, but loose dependencies do make pack testing a lot easier, in the cases where mods need to be disabled to track down a bug.
- This mod overall seems pretty unopinionated, so I don't see any issues for including this from a design perspective. That being said, if you are considering providing future contributions, with the goal of them being included in the pack, I would generally recommend you me give a heads-up before considering work on something (i.e. open an issue), to ensure the final result doesn't get rejected for any one of various reasons. Of course, if you're doing it to scratch an itch for your own private modifications, don't let me stop you.
- Use of the Hardcore Alchemy name: The way you're currently using it is OK, as it accurately describes what the mod is for. However, I'm a bit less chill about this when it comes to modpacks.

Also, next time you open a ticket, try to have one issue per ticket, except maybe if the issues are related. Thanks!

**Comment by asanetargoss** [comment]
Also...

> Code for Harvestcraft integration is included but the crop models are not working correctly in JER so the code is commented out.

Seems like a reasonable feature, even if the models aren't implemented yet.

**Comment by Invadermonky** [comment]
> Not sure what "Controllable Jump Boost" does

It's a way of controlling the jump height while under the effect of Potion of Leaping. The problem is the effect is negligible and it interferes with numerous other similar effects. I had a pain of a time tracking down this issue in the pack I develop for MC 1.12.2, so it's not a new bug but very few people know about it.


>  What is your preferred license for the mod?

I made the mod for myself and wasn't planning on releasing it, but I figured I would ask if you wanted it. I usually stick to MIT or WTFPL for the majority of the stuff I release, but if you wanted the code wholesale I'd leave the licensing up to you.


> I notice this mod adds compat for a variety of mods. The way it is written, it could crash if some of the referenced mods are not loaded, but I haven't tested this yet.

> That being said, if you are considering providing future contributions, with the goal of them being included in the pack, I would generally recommend you me give a heads-up before considering work on something (i.e. open an issue), to ensure the final result doesn't get rejected.

I made the mod for my own use so it does have a few issues like that. I figured I would ask if you were interested before devoting any amount of time to actually cleaning up the code base. I already have a published large-scale JER addon for 1.12.2 and I just threw this together over a few hours. It honestly took me longer to get the MDK running with dependencies than it did to actually code the mod.


> Use of the Hardcore Alchemy name: The way you're currently using it is OK, as it accurately describes what the mod is for. However, I'm a bit less chill about this when it comes to modpacks.

I 100% understand. The reason I used the Hardcore Alchemy name in this version was largely to keep it organized among the couple dozen mods I have in varying stages of development. If I was actually going to publish it, I would have released it as a 1.10 version of [Just Enough Magiculture](https://www.curseforge.com/minecraft/mc-mods/just-enough-magiculture).

Your ARR license on Hardcore Alchemy was the main reason I zipped the source rather than creating a repository.


> Also, next time you open a ticket, try to have one issue per ticket, except maybe if the issues are related. Thanks!

My bad.

### TLDR;
If you would like I can publish the code myself (under a different name to respect your ARR license), or you can have the code as-is and do whatever you want with it.

**Comment by asanetargoss** [comment]
With regard to licensing: The wording for the Hardcore Alchemy modpack license is not great for this situation, but I am generally OK with people making optional "plugins" intended for the modpack, under whatever license they want, provided that it would be otherwise legal. Uploading said plugins to mod sites, monetizing them, etc is not something that I *personally* oppose.

The thing to watch out for in this case is GPL mods, due to their copyleft requirements. The Hardcore Alchemy modpack contains a few of those. In fact, now that I look over your mod dependencies more carefully, it looks like your mod depends on Astral Sorcery, which is GPLv3, so in its current state [your mod has to be licensed under GPLv3](https://choosealicense.com/licenses/gpl-3.0/). However, at this point you *technically* could split the GPLv3 mod compat into a separate GPLv3 mod, and then relicense the rest to your choosing. As licensing is a personal choice, I leave it up to you to decide if this matters to you.

> I figured I would ask if you were interested before devoting any amount of time to actually cleaning up the code base.

That's good to hear. I do think that the the mod in its current state may be good enough for just my modpack's use case. Consider my feedback w/r/t classloading crashes to be optional, given the situation.

**Comment by Invadermonky** [comment]
It seems like the the best option would be to just publish the JER Integrations separately under a 1.10 version of JEM. It might take me a week or two before I have time to spend on the mod, so for now I'd say close this issue as complete (pertaining to the Jump Height bug) and I'll open another ticket once the mod is published.

**Comment by asanetargoss** [comment]
I will keep this ticket open for now, for tracking the jump height bug, and for future testing of the JER addon. I leave it up to you if you want to open additional ticket(s).

**Comment by Invadermonky** [comment]
JER Integration has been released.
https://www.curseforge.com/minecraft/mc-mods/just-enough-magiculture/files/5422901
